app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -59,84 +39,73 @@ public class EditPostActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_post);
 
-        // Configure window settings for fullscreen mode
-        requestWindowFeature(Window.FEATURE_NO_TITLE);
-        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
-                WindowManager.LayoutParams.FLAG_FULLSCREEN);
-        getSupportActionBar().hide();
+        DebugUtils.logUserFoldersAndPosts(User.getUser());
 
-        // Set the activity layout
-        setContentView(R.layout.activity_edit_post);
+        EditText linkToPost = findViewById(R.id.editTextLinkToPost);
+        EditText postTitle = findViewById(R.id.editTextPostTitle);
+        EditText postDescription = findViewById(R.id.editTextPostDescription);
+
+        Button buttonSaveChanges = findViewById(R.id.buttonSaveChanges);
+        Button buttonCancelChanges = findViewById(R.id.buttonCancelChanges);
+
+        String postId = getIntent().getStringExtra("postID");
+        String folderId = getIntent().getStringExtra("folderID");
 
-        // Prep vars
-        String currentFolderTitle;
         Post currentPost;
-        Folder currentFolder;
-        HashMap<Folder, FolderAccess> userFolders;
-        HashMap<String, Integer> folderIdNameMap = new HashMap<>();
-        EditText postNameField = findViewById(R.id.editTextNewName);
-        EditText postLinkField = findViewById(R.id.editTextNewLink);
-        Spinner spinner = findViewById(R.id.edit_post_spinner);
-        Button cancelBtn = findViewById(R.id.edit_post_cancel_btn);
-        Button saveBtn = findViewById(R.id.edit_post_save_button);
-        try {
-            currentPost = API.getPostById(getIntent().getIntExtra("postId", 0));
-            currentFolder = API.getFolderById(getIntent().getIntExtra("folderId", 0));
-            currentFolderTitle = String.format("(%d) %s", currentFolder.getId(), currentFolder.getName());
-            userFolders = API.getFoldersForUserId(PosteApplication.getCurrentUser().getId());
-            for (Folder folder: userFolders.keySet()) {
-                folderIdNameMap.put(String.format("(%d) %s", folder.getId(), folder.getName()), folder.getId());
+        Folder currentFolder = User.getUser().getFolder(folderId);
+        if (currentFolder == null){
+            currentPost = null;
+            Log.e("Error", "onCreate: Folder not found");
+        } else {
+            currentPost = User.getUser().getFolder(folderId).getPostFromFolder(postId);
```

Comment:
I like the chaining functionality here, but I wonder if a User method `getPost(String postId)` or `getPost(String folderId, String postId)` might be a simpler implementation? 

Since Posts are unique by ID, we don't *really* need to know the folder at this point; we could simply search for it through all folders. This is more performant, but the difference should be very small; up to you. But, abstracting the method call away might be nice. It would also give a way to handle a null folder OR post (because it can just return null if either post OR folder is null)

Reply:
I can do the search based on only postID... I'll implement it ASAP.
But Im confused on what you mean by abstracting the method call away. Can you elaborate? I'll shoot a message over Slack.

Reply:
I simply meant calling something like `user.getPostById(<postid>)`

Reply:
Has since been implemented